jax/scipy/linalg.py

```diff
@@ -379,3 +379,196 @@ def _pade13(A):
     U = np.dot(A,np.dot(A6, b[13]*A6 + b[11]*A4 + b[9]*A2) + b[7]*A6 + b[5]*A4 + b[3]*A2 + b[1]*ident)
     V = np.dot(A6, b[12]*A6 + b[10]*A4 + b[8]*A2) + b[6]*A6 + b[4]*A4 + b[2]*A2 + b[0]*ident
     return U,V
+
+@_wraps(scipy.linalg.expm_frechet,lax_description=textwrap.dedent("""\
+    Does not currently support the Scipy argument ``jax.numpy.asarray_chkfinite``, because `jax.numpy.asarray_chkfinite` does not exist at the moment. Does not support the ``method='blockEnlarge'`` argument.
+    """))
+def expm_frechet(A, E, *, method=None, compute_expm=True):
+    return _expm_frechet(A, E, method, compute_expm)
+
+def _expm_frechet(A, E, method=None, compute_expm=True):
+    A = np.asarray(A)
+    E = np.asarray(E)
+    if A.ndim != 2 or A.shape[0] != A.shape[1]:
+        raise ValueError('expected A to be a square matrix')
+    if E.ndim != 2 or E.shape[0] != E.shape[1]:
+        raise ValueError('expected E to be a square matrix')
+    if A.shape != E.shape:
+        raise ValueError('expected A and E to be the same shape')
+    if method is None:
+        method = 'SPS'
+    if method == 'SPS':
+        expm_A, expm_frechet_AE = expm_frechet_algo_64(A, E)
+    else:
+        raise ValueError('only method=\'SPS\' is supported')
+    expm_A, expm_frechet_AE = expm_frechet_algo_64(A, E)
+    if compute_expm:
+        return expm_A, expm_frechet_AE
+    else:
+        return expm_frechet_AE
+
+"""
+Maximal values ell_m of ||2**-s A|| such that the backward error bound
+does not exceed 2**-53.
+"""
+ell_table_61 = (
+        None,
+        # 1
+        2.11e-8,
+        3.56e-4,
+        1.08e-2,
+        6.49e-2,
+        2.00e-1,
+        4.37e-1,
+        7.83e-1,
+        1.23e0,
+        1.78e0,
+        2.42e0,
+        # 11
+        3.13e0,
+        3.90e0,
+        4.74e0,
+        5.63e0,
+        6.56e0,
+        7.52e0,
+        8.53e0,
+        9.56e0,
+        1.06e1,
+        1.17e1,
+        )
+        
+@jit
+def expm_frechet_algo_64(A, E):
+    ident = np.eye(*A.shape, dtype=A.dtype)
+    A_norm_1 = np_linalg.norm(A, 1)
+    """
+    Subset of the Maximal values ell_m of ||2**-s A||
+    such that the backward error bound does not exceed 2**-53.
```

Comment:
I'm guessing that we could use simpler algorithms for float32?

Reply:
Possibly. It could be argued that a different table should be used for float32 which could lead to the computation of lower Pade approximations. But I don't know what that table should be. 

If the question is whether Squaring and Scaling is the appropriate algorithm for float32, I don't know. S&S seems to be accepted widely. I have not examined Ward1977 or Sidje1998.